django-stubs/db/models/functions/comparison.pyi

```diff
@@ -16,8 +18,16 @@ class Collate(Func):
 class Greatest(Func): ...
 
 class JSONObject(Func):
-    def __init__(self, **fields: Any) -> None: ...
     output_field: ClassVar[JSONField]
+    def __init__(self, **fields: Any) -> None: ...
+    def as_sql(self, compiler: SQLCompiler, connection: BaseDatabaseWrapper, **extra_context: Any) -> _AsSqlType: ...  # type: ignore [override]
```

Comment:
Semi-incompatible override in Django itself, supporting other args only through kwargs